vscode-extension/src/server.ts

```diff
@@ -0,0 +1,222 @@
+      }
+      return [
+        getPropertyByKeyName(jsonDocument, 'wireit'),
+        getPropertyByKeyName(jsonDocument, 'scripts'),
+      ];
+    })();
+    this.#wireitProperty = wireit;
+    this.#wireitProperty?.children?.[1]?.children?.forEach((child) => {
+      const property = JsonProperty.fromAst(child);
+      if (property) {
+        this.#wireitConfigsByKey.set(property.key, property);
+      }
+    });
+    this.#scriptProperty = scripts;
+    this.#scriptProperty?.children?.[1]?.children?.forEach((child) => {
+      const property = JsonProperty.fromAst(child);
+      if (property) {
+        this.#scriptsByKey.set(property.key, property);
+      }
+    });
+    connection.console.log(util.inspect(this.#wireitConfigsByKey.entries()));
+  }
+
+  getDiagnostics(): Diagnostic[] {
+    return [
+      ...this.#checkThatWireitScriptsDeclaredInScriptsSection(),
+      ...this.#checkThatWireitScriptHasAtLeastOneOfCommandOrDependencies(),
+    ];
+  }
+
+  *#checkThatWireitScriptsDeclaredInScriptsSection(): IterableIterator<Diagnostic> {
+    const unclaimedWireitConfigs = new Map([...this.#wireitConfigsByKey]);
+
+    for (const prop of this.#scriptsByKey.values()) {
+      const hasWireitConfig = unclaimedWireitConfigs.delete(prop.key);
+      if (
+        hasWireitConfig &&
+        typeof prop.value === 'string' &&
+        prop.value.trim() !== 'wireit'
+      ) {
+        yield {
+          severity: DiagnosticSeverity.Error,
+          message: `This script is declared in the "wireit" section, but that won't have any effect unless this command is just "wireit"`,
+          source: 'wireit',
+          range: {
+            start: this.#textDocument.positionAt(prop.valueAst.offset),
+            end: this.#textDocument.positionAt(
+              prop.valueAst.offset + prop.valueAst.length
+            ),
+          },
+        };
+      }
+    }
+
+    for (const prop of unclaimedWireitConfigs.values()) {
+      yield {
+        severity: DiagnosticSeverity.Error,
+        message: `This script is declared in the "wireit" section, but not in the "scripts" section`,
+        source: 'wireit',
+        range: {
+          start: this.#textDocument.positionAt(prop.keyAst.offset),
+          end: this.#textDocument.positionAt(
+            prop.keyAst.offset + prop.keyAst.length
+          ),
+        },
+      };
+    }
+  }
+
+  *#checkThatWireitScriptHasAtLeastOneOfCommandOrDependencies(): IterableIterator<Diagnostic> {
+    for (const prop of this.#wireitConfigsByKey.values()) {
+      if (prop.valueAst.type !== 'object') {
+        continue;
+      }
+      const command = getPropertyByKeyName(prop.valueAst, 'command');
+      const dependencies = getPropertyByKeyName(prop.valueAst, 'dependencies');
+      if (command == null && dependencies == null) {
+        yield {
+          severity: DiagnosticSeverity.Error,
+          message: `Set either "command" or "dependencies", otherwise there's nothing for wireit to do.`,
```

Comment:
An empty `files` target is actually potentially useful too. You can use it kind of like a Bazel fileset that just affects cache invalidation:

```json
{
  "wireit": {
    "script1": {
      "dependencies": ["common-config-files"]
    },
    "script2": {
      "dependencies": ["common-config-files"]
    },
    "common-config-files": {
      "files": [
        "configA.js",
        "configB.json"
      ]
    }
  }
}
```

A little weird to do this as a _script_, since it's a no-op. But, just FYI. I'm also fine to ignore this for now.